08_flask_html_basics.md

```diff
@@ -0,0 +1,592 @@
+- 1xx informational response - the request was received by the server
+- 2xx successful - the request was processed successfully
+- 3xx redirect - the client is redirected to another resource or path
+- 4xx client error - the client's request is invalid
+- 5xx server error - there is an error with the server, when responding to an apparently correct request
+
+### Exercise 8.13
+
+Refresh the page and check out Flask's output in the VS Code terminal.
+
+- What status code is displayed for the `/` endpoint?
+- What status code is displayed for the image?
+- Why is the image status code different from the page's? You can check out [here](https://www.rfc-editor.org/rfc/rfc9110.html#name-304-not-modified) an explanation for the status code that you should be seeing.
+
+Try changing the HTTP operation in your code and refresh the page. What status code are you seeing now and why?
+
+<details markdown="1"><summary>Click here to reveal the answers</summary>
+
+- The status code for the `/` route should be `200` (OK), signifying that the request completed successfully
+- The status code for the image should be `200` for the first request, but upon refreshing the page it should be `304` (Not Modified). The client already has a copy of the image, so the server instructs the client to use the local copy instead of downloading a new one.
+- If you change the HTTP method in your endpoint to anything else but `GET`, you should receive a `405` (Method not allowed) response when refreshing the page.
+
+</details>
+
+
+### Exercise 8.14
+
+You will need to add a new endpoint to your application, mapping a route called `/add`. This will be responsible for adding an animal name to the existing list. The method should have a single parameter, which you can assume to be string.
+
+Consider the following questions:
+- Which HTTP method should you use for this endpoint? The requests will be made with the purpose of adding data to a "database"
+- What status code/action should the function return after adding the animal to the list? If the request is successful, the user should see the (updated) page on `/`
+
+Once you've come up with your answers, double-check with the answers below:
+
+<details markdown="1"><summary>Click here to reveal the answers</summary>
+
+- While we could use any HTTP operation, `POST` is the most appropriate one considering the conventions
+- Here are two ways of solving the problem:
+    - Return the same filled-in template after adding the animal to the list, so the response code should be `200`. However, you will notice that, after calling this endpoint, the URL will remain `/add` instead of `/`. This can be problematic, because when refreshing the page, the form could be resubmitted, resulting in additional POST requests with unintended consequences!
+    - Return a redirect to the `/` endpoint, so the response code would be one of `3xx`. This will instruct the broser to make a request to the URL for `/`. 
```

Comment:
```suggestion
    - Return a redirect to the `/` endpoint, so the response code would be one of `3xx`. This will instruct the browser to make a request to the URL for `/`. 
```